File: productHuntServer/src/main/java/com/nagarro/service/ReviewService.java
package com.nagarro.service;

import java.util.List;

import com.nagarro.model.product.Product;
import com.nagarro.model.product.Review;

public interface ReviewService {

	public List<Review> getReviewsByProduct(Product p);
	
	public List<Review> getNonApprovedReviews();
	
	public int avgRating(Long pId);
	
	public int countReviews(Long pId);
	
	public Review addReview(Review review);
	
	public void deleteReview(Long reviewId);
	
	public int numberOfReviews();
	
	public int approvedReview(Long reviewId);
			
	public Review getReview(Long reviewId);

}

File: productHuntFrontend/src/app/services/review.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class ReviewService {
  constructor(private _http: HttpClient) {}

  //add review
  public addReview(review: any) {
    return this._http.post(`${baseUrl}/review/`, review);
  }

  //non-approve reviews
  public reviews() {
    return this._http.get(`${baseUrl}/review/non-approve/`);
  }

  //delete review
  public deleteReview(reviewId: any) {
    return this._http.delete(`${baseUrl}/review/${reviewId}`);
  }

  //approve review
  public approveReview(reviewId: any) {
    return this._http.put(`${baseUrl}/review/${reviewId}`, reviewId);
  }

  //approved reviews by productId
  public getReviewsByProduct(pid: any) {
    return this._http.get(`${baseUrl}/review/product/approve/${pid}`);
  }

  //count reviews by productId
  public countReviews(pid: any) {
    return this._http.get(`${baseUrl}/review/count/${pid}`);
  }

  //avg rating
  public avgRating(pid: any) {
    return this._http.get(`${baseUrl}/review/avg/${pid}`);
  }

  //total reviews
  public numberOfReviews(): Observable<any> {
    return this._http.get(`${baseUrl}/review/`);
  }
}
